### ISSUE TITLE
build broken

### ISSUE THREAD

**Issue by shazqadeer** [p-org/P3#1]
I am trying to build P3 from a Windows 10 machine with VS 2017 installed.  I installed cmake 3.8.1 and then ran ".\build\build.cmd". I got an error:

C:\Users\qadeer\Work\P3>build\build.cmd

C:\Users\qadeer\Work\P3>set Configuration=Debug

C:\Users\qadeer\Work\P3>set PROFILE=MSVC2017

C:\Users\qadeer\Work\P3>set Platform=x64

C:\Users\qadeer\Work\P3>set Toolset=v140

C:\Users\qadeer\Work\P3>set CMAKE_GENERATOR=Visual Studio 15 2017 Win64

C:\Users\qadeer\Work\P3>set CMAKE_OPTIONS=-DUSE_RUNTIME_DLL=ON

C:\Users\qadeer\Work\P3>set CMAKE_BUILD_TYPE=Debug

C:\Users\qadeer\Work\P3>set BUILD_TRIPLE=MSVC2017-x64-Debug

C:\Users\qadeer\Work\P3>set INSTALL_DIR=../MSVC2017-x64-Debug

C:\Users\qadeer\Work\P3>set BUILD_DIR=MSVC2017-x64-Debug-build

C:\Users\qadeer\Work\P3>mkdir "MSVC2017-x64-Debug-build"
A subdirectory or file MSVC2017-x64-Debug-build already exists.

C:\Users\qadeer\Work\P3>cd "MSVC2017-x64-Debug-build"

C:\Users\qadeer\Work\P3\MSVC2017-x64-Debug-build>cmake -G "Visual Studio 15 2017 Win64" -T "v140" ../.. "-DCMAKE_BUILD_TYPE=Debug" -DUSE_RUNTIME_DLL=ON
CMake Error: The source directory "C:/Users/qadeer/Work" does not appear to contain CMakeLists.txt.

It looks like I should be running build.cmd from the build folder.  In that case, the build instruction should be updated.

Next, I tried running build.cmd from the build folder.  I got the following output:

C:\Users\qadeer\Work\P3\build>build.cmd

C:\Users\qadeer\Work\P3\build>set Configuration=Debug

C:\Users\qadeer\Work\P3\build>set PROFILE=MSVC2017

C:\Users\qadeer\Work\P3\build>set Platform=x64

C:\Users\qadeer\Work\P3\build>set Toolset=v140

C:\Users\qadeer\Work\P3\build>set CMAKE_GENERATOR=Visual Studio 15 2017 Win64

C:\Users\qadeer\Work\P3\build>set CMAKE_OPTIONS=-DUSE_RUNTIME_DLL=ON

C:\Users\qadeer\Work\P3\build>set CMAKE_BUILD_TYPE=Debug

C:\Users\qadeer\Work\P3\build>set BUILD_TRIPLE=MSVC2017-x64-Debug

C:\Users\qadeer\Work\P3\build>set INSTALL_DIR=../MSVC2017-x64-Debug

C:\Users\qadeer\Work\P3\build>set BUILD_DIR=MSVC2017-x64-Debug-build

C:\Users\qadeer\Work\P3\build>mkdir "MSVC2017-x64-Debug-build"

C:\Users\qadeer\Work\P3\build>cd "MSVC2017-x64-Debug-build"

C:\Users\qadeer\Work\P3\build\MSVC2017-x64-Debug-build>cmake -G "Visual Studio 15 2017 Win64" -T "v140" ../.. "-DCMAKE_BUILD_TYPE=Debug" -DUSE_RUNTIME_DLL=ON
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_C_COMPILER could be found.

CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_CXX_COMPILER could be found.

-- Configuring incomplete, errors occurred!
See also "C:/Users/qadeer/Work/P3/build/MSVC2017-x64-Debug-build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/qadeer/Work/P3/build/MSVC2017-x64-Debug-build/CMakeFiles/CMakeError.log".

C:\Users\qadeer\Work\P3\build\MSVC2017-x64-Debug-build>cmake --build . --config "Debug"
Microsoft (R) Build Engine version 15.1.1012.6693
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: ALL_BUILD.vcxproj

C:\Users\qadeer\Work\P3\build\MSVC2017-x64-Debug-build>cmake "-DCMAKE_INSTALL_PREFIX=../MSVC2017-x64-Debug" "-DBUILD_TYPE=Debug" -P cmake_install.cmake
CMake Error: Error processing file: cmake_install.cmake

Finally, the repository does not appear to have a master branch.  As a result the clone gives an empty enlistment and I have to switch to the dev branch.

**Comment by paulthomson** [comment]
I think using a VS command prompt might fix the issue.

**Comment by shazqadeer** [comment]
I am running from Developer Command Prompt for VS 2017.

**Comment by paulthomson** [comment]
Perhaps the toolset specified is not installed. You could try v141. This is just a guess though.

**Comment by shazqadeer** [comment]
Yes, that was the issue.  When I changed Toolset in build.cmd to v141, the build succeeded. I suppose either build.cmd needs to be parameterized or we specify the dependency explicitly.  I believe that v141 is what is installed by default when you install VS 2017.
